app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/HorizontalSnapScrollState.kt

```diff
@@ -0,0 +1,139 @@
+        }
+    }
+
+    fun isRoomsCountInitialized() = roomsCount != NOT_INITIALIZED
+
+    val maxColumnIndex
+        get() = max(roomsCount - displayColumnCount, 0)
+
+    /**
+     * Returns the column index calculated based on the given [scrollX] value
+     * and the current state of [HorizontalSnapScrollState].
+     */
+    fun calculateOnTouchColumnIndex(scrollX: Int): Int {
+        val relativeDistance = scrollX - xStart
+        var columnIndex = activeColumnIndex
+
+        logging.d(
+            LOG_TAG,
+            """column width: $columnWidth, scrollX: $scrollX, distance: $relativeDistance, activeColumnIndex: $activeColumnIndex"""
+        )
+
+        var columnIndexDelta = calculateRelativeColumnIndexDelta(relativeDistance)
+
+        if (displayColumnCount <= 1 && checkPortraitModeScrollDistance(relativeDistance)) {
+            columnIndexDelta = if (relativeDistance > 0) 1 else -1
+        }
+
+        logging.d(LOG_TAG, "column distance: ${abs(columnIndexDelta)}")
+
+        columnIndex -= columnIndexDelta
+
+        columnIndex = max(columnIndex, 0)
+        columnIndex = min(columnIndex, roomsCount - displayColumnCount)
+
+        return columnIndex
+    }
+
+    /**
+     * Checks if a given scroll motion is far enough for a scroll
+     * to the next column in portrait mode, based on the [columnWidth]
+     */
+    fun checkPortraitModeScrollDistance(relativeDistance: Int) =
+        abs(relativeDistance) > columnWidth * SCROLL_THRESHOLD_FACTOR
+
+    /**
+     * Calculates the relative Column Delta based on the [columnWidth]
+     */
+    fun calculateRelativeColumnIndexDelta(relativeDistance: Int): Int {
+        val absoluteColumnDelta = abs(relativeDistance.toFloat()) / columnWidth
+        if (relativeDistance > 0)
+            return absoluteColumnDelta.roundToInt()
+        else {
+            return -(absoluteColumnDelta).roundToInt()
+        }
+    }
+
+    /**
+     * Returns the horizontal position calculated based on the given [columnIndex] and
+     * the current value of [columnWidth].
+     */
+    fun calculateScrollToXPosition(columnIndex: Int): Int {
+        val constraintColumnIndex = when {
+            columnIndex < 0 -> 0
+            columnIndex > maxColumnIndex -> maxColumnIndex
+            else -> columnIndex
+        }
+        return constraintColumnIndex * columnWidth
+    }
+
+    /**
+     * Returns the column width calculated based on the given [measuredWidth]
+     * and the current value of [displayColumnCount].
+     */
+    fun calculateColumnWidth(measuredWidth: Int): Int {
+        return (measuredWidth.toFloat() / displayColumnCount).roundToInt()
+    }
+
+    /**
+     * Checks if the velocity of a Fling event was big enough to initiate a 'flingscroll'
+     * @param normalizedVelocity the X Velocity of the Fling independent from the display pixel-density
+     */
+    fun checkFlingVelocity(normalizedVelocity: Float) =
```

Comment:
The `checkFlingVelocity` and `checkScrollDistance` functions and the corresponding constants do not really belong in this class because they do not interact with any of the class properties. This can also be seen in the unit test where a state instance is created but not used.

Instead I would add the functions to the end of `HorizontalSnapScrollView.kt` (similar as  `ViewGroup.firstChild`) and make them accessible for testing via `@VisibleForTesting`.